src/components/Thumbnail.js

```diff
@@ -0,0 +1,53 @@
+import classnames from 'classnames';
+import { toChildArray } from 'preact';
+
+import { Spinner } from './Spinner';
+
+/**
+ * @typedef {import('preact').ComponentChildren} Children
+ *
+ * @typedef ThumbnailProps
+ * @prop {Children|null} [children] - Thumbnail content (typically an img)
```

Comment:
I didn't start with the `null`. I added it because when I tried to use this component without it in the `BookPicker` in `lms`, the following raised a type-checking error (this isn't the exact code, but the pattern is representative):

```
<Thumbnail>
  { ifThereIsAnImageAvailable && <img src ... /> }
</Thumbnail>
```

This caused a type-checking complaint because `null` is not assignable to `Children | undefined`.

Any words of wisdom?

Reply:
I don't see any type errors:
```
/** @param {boolean|null| undefined} show */
const MyComponent = show => {
  return <Thumbnail>{show && <span>spam</span>}</Thumbnail>;
};
```